tests/test_execute.py

```diff
@@ -58,36 +58,26 @@ def doctree(
         shutil.rmtree(tree)
 
 
-def test_basic(doctree):
+def test_basic(doctree, file_regression):
     source = """
     .. jupyter-execute::
 
         2 + 2
     """
     tree = doctree(source)
     (cell,) = tree.traverse(JupyterCellNode)
-    assert cell.attributes["code_below"] is False
-    assert cell.attributes["hide_code"] is False
-    assert cell.attributes["hide_output"] is False
-    assert cell.attributes["linenos"] is False
-    assert cell.children[0].rawsource.strip() == "2 + 2"
-    assert cell.children[1].rawsource.strip() == "4"
+    file_regression.check(cell.pformat(), extension=".xml")
 
 
-def test_basic_old_entrypoint(doctree):
+def test_basic_old_entrypoint(doctree, file_regression):
     source = """
     .. jupyter-execute::
 
         2 + 2
     """
     tree = doctree(source, entrypoint="jupyter_sphinx.execute")
     (cell,) = tree.traverse(JupyterCellNode)
-    assert cell.attributes["code_below"] is False
-    assert cell.attributes["hide_code"] is False
-    assert cell.attributes["hide_output"] is False
-    assert cell.attributes["linenos"] is False
-    assert cell.children[0].rawsource.strip() == "2 + 2"
-    assert cell.children[1].rawsource.strip() == "4"
+    file_regression.check(cell.pformat(), extension=".xml")
```

Comment:
here is how we use `file_regression`. we pass it some data, and it checks the content of that data against a file that is written to disk in the repository. If they don't match, then the test will fail